decoder.go

```diff
@@ -16,9 +16,60 @@ const (
 	defaultMaxSize = 16000
 )
 
+type option[T Decoder | Encoder] func(d *T)
```

Comment:
Made this private to prevent people from creating their own.

Reply:
good idea